File: extensions/aether-space/kotlin/src/main/kotlin/space/SpaceService.kt
package space

import java.util.UUID

interface ISpaceService {
    suspend fun dropBreadcrumb(breadcrumb: SpaceBreadcrumb): SpaceBreadcrumb
    suspend fun scan(geoHash: String, radiusKm: Double = 5.0): List<SpaceBreadcrumb>
    suspend fun pinBreadcrumb(breadcrumbId: UUID, spaceId: UUID): SpaceBreadcrumb
    suspend fun unpinBreadcrumb(breadcrumbId: UUID, spaceId: UUID): SpaceBreadcrumb
    suspend fun deleteBreadcrumb(breadcrumbId: UUID, requesterId: UUID): Boolean
    suspend fun getById(breadcrumbId: UUID): SpaceBreadcrumb?
    suspend fun listBySpace(spaceId: UUID, limit: Int = 50, offset: Int = 0): List<SpaceBreadcrumb>
    suspend fun react(breadcrumbId: UUID, userId: UUID, reaction: String): Int
}

class SpaceServiceImpl : ISpaceService {

    override suspend fun dropBreadcrumb(breadcrumb: SpaceBreadcrumb): SpaceBreadcrumb {
        TODO("not implemented")
    }

    override suspend fun scan(geoHash: String, radiusKm: Double): List<SpaceBreadcrumb> {
        TODO("not implemented")
    }

    override suspend fun pinBreadcrumb(breadcrumbId: UUID, spaceId: UUID): SpaceBreadcrumb {
        TODO("not implemented")
    }

    override suspend fun unpinBreadcrumb(breadcrumbId: UUID, spaceId: UUID): SpaceBreadcrumb {
        TODO("not implemented")
    }

    override suspend fun deleteBreadcrumb(breadcrumbId: UUID, requesterId: UUID): Boolean {
        TODO("not implemented")
    }

    override suspend fun getById(breadcrumbId: UUID): SpaceBreadcrumb? {
        TODO("not implemented")
    }

    override suspend fun listBySpace(spaceId: UUID, limit: Int, offset: Int): List<SpaceBreadcrumb> {
        TODO("not implemented")
    }

    override suspend fun react(breadcrumbId: UUID, userId: UUID, reaction: String): Int {
        TODO("not implemented")
    }
}

File: extensions/aether-space/swift/Sources/AetherSpace/SpaceService.swift
import Foundation

public protocol SpaceServiceProtocol {
    func dropBreadcrumb(_ breadcrumb: SpaceBreadcrumb) async throws -> SpaceBreadcrumb
    func scan(geoHash: String, radiusKm: Double) async throws -> [SpaceBreadcrumb]
    func pinBreadcrumb(breadcrumbId: UUID, spaceId: UUID) async throws -> SpaceBreadcrumb
    func unpinBreadcrumb(breadcrumbId: UUID, spaceId: UUID) async throws -> SpaceBreadcrumb
    func deleteBreadcrumb(breadcrumbId: UUID, requesterId: UUID) async throws -> Bool
    func getById(_ breadcrumbId: UUID) async throws -> SpaceBreadcrumb?
    func listBySpace(_ spaceId: UUID, limit: Int, offset: Int) async throws -> [SpaceBreadcrumb]
    func react(breadcrumbId: UUID, userId: UUID, reaction: String) async throws -> Int
}

public class SpaceService: SpaceServiceProtocol {

    public init() {}

    public func dropBreadcrumb(_ breadcrumb: SpaceBreadcrumb) async throws -> SpaceBreadcrumb {
        fatalError("not implemented")
    }

    public func scan(geoHash: String, radiusKm: Double = 5.0) async throws -> [SpaceBreadcrumb] {
        fatalError("not implemented")
    }

    public func pinBreadcrumb(breadcrumbId: UUID, spaceId: UUID) async throws -> SpaceBreadcrumb {
        fatalError("not implemented")
    }

    public func unpinBreadcrumb(breadcrumbId: UUID, spaceId: UUID) async throws -> SpaceBreadcrumb {
        fatalError("not implemented")
    }

    public func deleteBreadcrumb(breadcrumbId: UUID, requesterId: UUID) async throws -> Bool {
        fatalError("not implemented")
    }

    public func getById(_ breadcrumbId: UUID) async throws -> SpaceBreadcrumb? {
        fatalError("not implemented")
    }

    public func listBySpace(_ spaceId: UUID, limit: Int = 50, offset: Int = 0) async throws -> [SpaceBreadcrumb] {
        fatalError("not implemented")
    }

    public func react(breadcrumbId: UUID, userId: UUID, reaction: String) async throws -> Int {
        fatalError("not implemented")
    }
}
